core/src/main/java/com/exadel/aem/toolkit/core/injectors/EToolboxListInjector.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.models.spi.DisposalCallbackRegistry;
+import org.apache.sling.models.spi.Injector;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.day.cq.commons.jcr.JcrConstants;
+
+import com.exadel.aem.toolkit.api.annotations.injectors.EToolboxList;
+import com.exadel.aem.toolkit.core.lists.utils.ListHelper;
+
+/**
+ * Injects EToolbox Lists into Sling models obtained via {@code ResourceResolver} instance
+ * @see ListHelper
+ * @see Injector
+ */
+@Component(service = Injector.class,
+    property = Constants.SERVICE_RANKING + ":Integer=" + InjectorConstants.SERVICE_RANKING
+)
+public class EToolboxListInjector implements Injector {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EToolboxListInjector.class);
+
+    public static final String NAME = "eak-toolbox-list-injector";
+
+    /**
+     * Retrieves the name of the current instance
+     * @return String value
+     * @see Injector
+     */
+    @Nonnull
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    /**
+     * Attempts to inject list entries into the given adaptable
+     * @param adaptable        A {@link SlingHttpServletRequest} or a {@link Resource} instance
+     * @param name             Name of the Java class member to inject the value into
+     * @param type             Type of receiving Java class member
+     * @param element          {@link AnnotatedElement} instance that facades the Java class member allowing to retrieve
+     *                         annotation objects
+     * @param callbackRegistry {@link DisposalCallbackRegistry} object
+     * @return The value to inject, or null in case injection is not possible
+     * @see Injector
+     * @see ListHelper
+     */
+    @Override
+    public Object getValue(
+        @Nonnull Object adaptable,
+        String name,
+        @Nonnull Type type,
+        @Nonnull AnnotatedElement element,
+        @Nonnull DisposalCallbackRegistry callbackRegistry) {
+
+        EToolboxList annotation = element.getDeclaredAnnotation(EToolboxList.class);
+        if (annotation == null) {
+            return null;
+        }
+
+        ResourceResolver resourceResolver = InjectorUtils.getResourceResolver(adaptable);
+        if (resourceResolver == null) {
+            return null;
+        }
+
+        if (annotation.value().isEmpty()) {
+            LOG.debug(InjectorConstants.EXCEPTION_EMPTY_ANNOTATION_VALUE);
+            return null;
+        }
+
+        if (!annotation.keyProperty().isEmpty()) {
+            return InjectorUtils.isValidMapKeyType(type, String.class)
+                ? ListHelper.getMap(resourceResolver, annotation.value(), annotation.keyProperty(), getTypeArgument(type, 1))
+                : null;
```

Comment:
I think we shouldn't be so cruel to the users :) 
If we can theoretically return a valid list, we should definitely do so)
If the user specifies the keyProperty and uses a non-map type, we can simply ignore the keyProperty (and maybe throw a warning)

Reply:
Agreed. And don't forget we can always inject anything at all into `Object`
In fact, there are variants here. If a list of resources has a property "id" (or "code" etc.), of type Long, technically we can easily have a Map<Long, Something>, and it will work.
I only have a doubt that we do need all of these variants

Reply:
> In fact, there are variants here. If a list of resources has a property "id" (or "code" etc.), of type Long, technically we can easily have a Map<Long, Something>, and it will work.
> I only have a doubt that we do need all of these variants.

Should I add Map<Long, T> and Map<Integer, T> to the injector?
Will this be enough?

Reply:
Added variants with Object and moved the keyProperty check inside the map check so that it would be possible to inject in cases when the user specifies keyProperty not with Map type.